Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -205,28 +207,40 @@ void RenderTheme::adjustStyle(RenderStyle& style, const Element* element, const
     else if (style.display() == DisplayType::ListItem || style.display() == DisplayType::Table)
         style.setEffectiveDisplay(DisplayType::Block);
 
-    if (userAgentAppearanceStyle && isControlStyled(style, *userAgentAppearanceStyle)) {
-        switch (appearance) {
-        case StyleAppearance::Menulist:
-            appearance = StyleAppearance::MenulistButton;
-            break;
-        default:
-            appearance = StyleAppearance::None;
-            break;
-        }
-
-        style.setUsedAppearance(appearance);
+    bool hasDropdownIndicator = false;
+    switch (autoAppearance) {
+#if ENABLE(DATALIST_ELEMENT)
+    case StyleAppearance::ListButton:
+#endif
+    case StyleAppearance::Menulist:
+    case StyleAppearance::MenulistButton:
+        hasDropdownIndicator = true;
+        break;
+    default:
+        break;
     }
 
-    if (appearance == StyleAppearance::SearchField && searchFieldShouldAppearAsTextField(style)) {
-        appearance = StyleAppearance::TextField;
-        style.setUsedAppearance(appearance);
-    }
+    if (hasDropdownIndicator) {
+        if (userAgentAppearanceStyle && isControlStyled(style, *userAgentAppearanceStyle)) {
```

Comment:
You can entirely get rid of `isControlStyled` and inline most of the relevant logic here (e.g. only the select cases)